File: pgm/ABC_C#/ABC_159_B.cs
using System;
// using System.Collections.Generic;

class Program
{
    static bool checkPal(string s)
    {
        char[] arr = s.ToCharArray();
        Array.Reverse(arr);
        return s == new string(arr);
    }

    static void Main()
    {
        String S = Console.ReadLine();
        String str = S.Substring(0, S.Length / 2);
        Console.WriteLine(checkPal(S) && checkPal(str) ? "Yes" : "No");
    }
}

File: pgm/ABC_Visual_Basic/ABC_159_B.vb
' Imports System.Collections.Generic
' Imports System.Text.RegularExpressions

Class Main
    Public Shared Function checkPal(ByVal s As String) As Boolean
        return s = StrReverse(s)
    End Function

    Public Shared Sub Main()
        Dim S As String = Console.ReadLine()
        Dim str As String = S.Substring(0, Math.floor(S.Length / 2))
        Console.WriteLine(If(checkPal(S) AndAlso checkPal(str),"Yes", "No"))
    End Sub
End Class
